### DISCUSSION TITLE
Option to transfer all the aliases when deleting a mailbox

### DISCUSSION THREAD

**Discussion by nguyenkims** [simple-login/app#410] [Feature Requests]
When deleting a mailbox, offer an option to transfer all the aliases that go to that mailbox into another mailbox.

Moved from https://trello.com/c/hE0bR2Wi/62-when-deleting-a-mailbox-offer-an-option-to-transfer-all-the-aliases-that-go-to-that-mailbox-into-another-mailbox

**Comment by WhamboMPS** [comment]
I would like to offer an important use case for this feature.

My wife and I use one SimpleLogin account which has two mailboxes, one that is her personal email address and one that is mine. The aliases assigned to these mailboxes cover everything from the trivial (marketing emails) to essential (utility bills). If one of us were to suddenly become disabled or die, it would be important to switch all of the aliases currently going to the sick spouse immediately over to the healthy one.

I think that this is an important feature and would ask that it be prioritized.

**Reply by nguyenkims** [reply]
Thanks for the info. I'll make sure to prioritise this feature in our next version. It shouldn't be too complex to implement.

**Reply by WhamboMPS** [reply]
Is this expected to make it into the Q2-2022 release?

Also, you _may_ be able to collapse issue 407 and/or 839 into this one.

**Reply by nguyenkims** [reply]
407 and 839 are actually more tricky as we need to find a way to bulk select aliases in order to make bulk edition. This'd require to convert the web app to a SPA (single page app) and this is actually a huge task. We plan to work on this conversion in Q2 and once it's done, we can start working on 407 and 839.

**Reply by M4357R0-H4Z3** [reply]
Many thanks for your efforts. Looking for this feature as well. I have a need to change over 200 aliases to a new mailbox. Perhaps You could implement this in phases? In my use case I only need all the address to (auto) change to my current default mailbox from the old one when I delete it.
IMO this is an essential feature for customers already with simple logon to migrate from current 'other' email services to proton mail :)

**Comment by WhamboMPS** [comment]
I have a potential error condition to suggest if this feature is implemented.

Assume the SimpleLogin account has two real email addresses registered with it, john@gmail.com and mary@gmail.com. Next assume that John dies. The goal of this feature is to create an easy method for all email currently being redirected to John to instead be sent to Mary. Fine so far...

But... what if a specific alias currently redirects to _both_ of the two registered real email addresses (john@gmail.com _and_ mary@gmail.com). If you do a simple substitution of the real address in the "Mailbox" field from john@gmail.com to mary@gmail.com then the "Current mailbox" field of the Alias card would read

`mary@gmail.com, mary@gmail.com`

And I suspect that internal duplication like this would throw an error. And because it's a batch process the whole process might fail.

Moral to the story is that when changing the destination real address to support this feature you should query the "Current Mailbox" field _for every alias_ to remove any duplication of real email address.

**Comment by mxng** [comment]
Still waiting for this, as I have an old mailbox with over 400 aliases and a new one that I'm primarily using with over 180 aliases. I can't just change the mailbox of the old one to the new address because the new one is already in my account. Would be nice if I can just delete the old mailbox and let the 400 over aliases add to the mailbox with over 180 aliases.

**Comment by nguyenkims** [comment]
This feature is now present in the latest version.

**Reply by TheCrazyMax** [reply]
Can you move it Done on the Roadmap as well ? :) It is still being shown as To Do. Thanks :)

**Comment by tomaspapi** [comment]
Hi. I see this feature was shipped. Does it cover bulk migrating all aliases into another mailbox, or do we have to do it one by one on the “Aliases” tab? If bulk is possible, can anyone please share steps? Thank you.

**Comment by jeremiah-rs** [comment]
Is it possible to bulk transfer aliases to a different account? I want to move them to my Proton account if possible.

**Reply by Gnemesis42a** [reply]
I found a way to get around the lack of a direct bulk transfer. 
1. Add the mailbox that you want to be the destination for these aliases.
2. Delete the mailbox that you want to drop as the recipient from the list of mailboxes under Settings->Aliases. During that step, you will be asked to transfer the recipient to another mailbox ELSE THOSE ALIASES WILL BE DELETED. Elect the new mailbox and all will be under the new mailbox address.
